features/192-asset-gear/index.md

```diff
@@ -0,0 +1,640 @@
+# Asset Gear
+
+## Goals
+
+- Store user-generated asset to fulfill application needs.
+  - Photos and short videos taken by the application.
+  - Images and short videos chosen from the library.
+  - Artwork by the user.
+  - Attachments in a chat application.
+  - Documents of reasonable size.
+- Process image asset on the fly via query string.
+- Access control via presigned URL
+
+## Non-goals
+
+- Video live streaming such as Facebook Live.
+- File sharing application like Imgur.
+- Static website hosting. Another gear is responsible for this.
+
+## Layout
+
+The asset name is in the following format.
+
+```
+/<app-id>/<random-string>[.<ext>]
```

Comment:
Sorry can't imagine I forgot it in the first place, do we support custom domain name for asset storage? That would be a MUST ask by end users...

Reply:
An example URL of an asset named `myimage.png` looks like `https://myapp.skygearapps.com/_asset/get/myimage.png`. When we support custom domain, the URL becomes `https://myapp.com/_asset/get/myimage.png`, just like any other endpoints.